ExploreASL_Initialize.m

```diff
@@ -697,111 +639,242 @@
+        fullfile('External','SPMmodified','matlabbatch'),...
+        fullfile('External','SPMmodified','xASL'),...
+        fullfile('External','SPMmodified','toolbox','cat12'), ...
+        fullfile('External','SPMmodified','toolbox','LST'), ...
+        fullfile('External','SPMmodified','toolbox','OldNorm') ...
+        genpath(fullfile('External','bids-matlab'))};
+    % Iterate over subfolders which should be added
+    for ii=1:length(subfolders_to_add)
+        addpath(fullfile(MyPath,subfolders_to_add{ii}));
+    end
+    
+end
 
 %% -----------------------------------------------------------------------
-%% 3) Print warnings
-fprintf('\n%s\n\n','---------------------------------------------');
+%% Define input parser
+function p = inputParsing(varargin)
 
-field_symbol = {'subject_regexp'};
+    % Initialize input parser
+    p = inputParser;
+    
+    % Define valid input variables
+    validDataParPath = @(variable) ischar(variable) || isempty(variable);
+    validImportArray = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
+    validProcessArray = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
+    validSkipPause = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
+    validiWorker = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
+    validnWorkers = @(variable) ischar(variable) || isempty(variable) || isnumeric(variable);
+    
+    % Define defaults
+    defaultDataParPath = [];
+    defaultImportArray = [0 0 0 0];
+    defaultProcessArray = [0 0 0];
+    defaultSkipPause = 0;
+    defaultiWorker = 1;
+    defaultnWorkers = 1;
+    
+    % Add definitions to the input parser
+    addOptional(p, 'DataParPath', defaultDataParPath, validDataParPath);
+    addOptional(p, 'ImportArray', defaultImportArray, validImportArray);
+    addOptional(p, 'ProcessArray', defaultProcessArray, validProcessArray);
+    addOptional(p, 'SkipPause', defaultSkipPause, validSkipPause);
+    addOptional(p, 'iWorker', defaultiWorker, validiWorker);
+    addOptional(p, 'nWorkers', defaultnWorkers, validnWorkers);
+    
+    % Parse input
+    parse(p,varargin{:});
 
-for iField=1:length(field_symbol)
-    if ~isfield(x,field_symbol{iField})
-        warning(['x.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
-    end
 end
 
-if ~isfield(x,'D')
-    warning('x.D didn''nt exist');
-else
-    field_symbol = {'ROOT'};
-    for iField=1:length(field_symbol)
-        if ~isfield(x.D,field_symbol{iField})
-            warning(['x.D.' field_symbol{iField} ' was not defined in DATA_PAR.m!'])
-        end
+%% -----------------------------------------------------------------------
+%% Convert parsed input
+function parameters = convertParsedInput(parameters)
+
+    % Check if inputs are empty or chars
+    if isempty(parameters.DataParPath),     parameters.DataParPath = '';                                    end
```

Comment:
`,` should be `;` though `,` is perhaps also possible, `;` makes more sense right?

Reply:
I always use `,`. Saw both options here https://stackoverflow.com/questions/27561881/one-liner-for-if-then Seems like a style-wise choice.